### PR TITLE
Pdf print

### PR DESCRIPTION
**Pull request by skilef**
# Export to PDF
Solves Issue #8

Added the ability to export the calendar as a PDF file.
The PDF file is being downloaded immediately.

## Technical stuff
* The PDF is created with **jsPDF** and **jsPDF-AutoTable** (plugin)
* The PDF is **textual** and not an image
* UTF and RTL support is limited
  * Special Javascript font files are also included for this purpose
  * Text direction was reversed several times
* I also tried working with other libraries but they had their own problems
  * **html2pdf.js** actually worked well but it generated an image based PDF (using **html2canvas**) and the file was really heavy

#### Not Logged-In
image
image

#### Logged-In
image

## PDF Calendar
image

### CODE REVIEW

**Comment by michael-maltsev** [comment]
Very nice, thank you for the contribution! It looks good overall, but there are a couple of things to be taken care of before it gets merged.

* Functionality: I tested it briefly, and the most noticeable issue is that the table is limited to the fixed times of 8:30-17:30 and days of Sunday-Thirsday. CheeseFork supports any time and any day of week. You can see that e.g. with the 394902 course which has several late courses, or by dragging a custom event to a different time or day.

* User interface: I don't like merging "Share" and "Export" to a single menu. Let's keep "Share" ("שיתוף מערכת") as is, and have an "Export" ("ייצוא") item with two options.

* Code: I'll leave a couple of minor comments in a moment.

File: deploy/cheesefork.js [hunk 1]
```diff
@@ -935,14 +941,16 @@
             $('#top-navbar-login').addClass('d-none');
             $('#top-navbar-logout').removeClass('d-none')
                 .find('a').attr('data-original-title', 'מחובר בתור: ' + user.displayName);
-            $('#top-navbar-share').find('a').removeClass('disabled').tooltip('disable');
+            $('#top-navbar-share-link').find('a').removeClass('disabled').tooltip('disable');
+            $('#top-navbar-share-link').css("display", "block");
```
Comment (michael-maltsev) [inline deploy/cheesefork.js:945]: Use `addClass('d-none')` and `removeClass('d-none')` for consistency.

File: deploy/components/course-calendar/course-calendar.js [hunk 2]
```diff
@@ -980,6 +980,171 @@ var CourseCalendar = (function () {
...
+            theme: 'grid',
+            startY: 40,
+            styles: {
+                font: "Rubik",
+                halign: "center",
+                lineWidth: 0.1,
+            },
+        });
+    
+        pdf.save("Calendar.pdf");
```
Comment (michael-maltsev) [inline deploy/components/course-calendar/course-calendar.js:1145]: We can use a more "friendly" file name for the PDF, as we do now for the ics file (e.g. `winter-2020-2021.pdf`).

File: deploy/cheesefork.css [hunk 3]
```diff
@@ -25,6 +25,10 @@
     visibility: hidden !important;
 }
 
+#top-navbar-share-link {
+    display: none;
```
Comment (michael-maltsev) [inline deploy/cheesefork.css:29]: It's unnecessary for the other menu items. For consistency, please implement the new one in the same way.

File: deploy/index.html [hunk 4]
```diff
@@ -292,6 +300,11 @@ <h3 id="sharing-user-known" class="d-none">
 <script src="modules/intro.js/intro.min.js"></script>
 <script src="modules/reparse.js"></script>
 <script src="modules/boolean-expression.js"></script>
+<script src="https://cdn.jsdelivr.net/npm/es6-promise@4/dist/es6-promise.min.js"></script>
```
Comment (michael-maltsev) [inline deploy/index.html:303]: All other modules are kept locally. Even thought it's a small file, please create a local `modules/es6-promise` folder for it and import it locally.

**Review by michael-maltsev — CHANGES_REQUESTED** [review]

File: deploy/index.html (hunk 4 above)
Comment (skilef) [reply deploy/index.html:303]: Oh sorry it's my fault, forgot to download it.

**Comment by skilef** [comment]
# PDF feature with all fixes
I fixed the things you @michael-maltsev mentioned in the new commits
* Also now the PDF looks more like the original calendar
* And the code is better documented

image

**Comment by skilef** [comment]
I forgot to set the column sizes, so please wait with merging

**Comment by skilef** [comment]
All fixed.
image

Check it out

**Comment by skilef** [comment]
All done, you can check it out
image

**Comment by michael-maltsev** [comment]
Thanks. Issues that I've seen from a quick glance:

* Text of small events (1/2 hour) is not displayed:
image

* Also, text of larger events (1 hour) is not vertically aligned:
image

* The sharing button is gone, at least when not signed in:
image

Also, could you rebase the code on the most resent commit?

**Comment by skilef** [comment]
Rebased everything to a single commit and fixed the things you mentioned.

**Comment by michael-maltsev** [comment]
I didn't mean squashing the commits, I meant rebasing to update to the recent changes of the repository. Not a big deal, I can do it myself later.

More importantly, there are still issues unfortunately:

* Overlapping events aren't handled properly:

image

* English text is reversed. You can see that in the image above as well.

* A single large event screws up the PDF, example:

image

I didn't spend a lot of time testing it. Please, next time, if possible, play with the solution a bit to see if there are no other things to be done, to save on the ping-pong.

Also, the tooltip from a disabled sharing button is now gone:

image

**Comment by skilef** [comment]
Ok, sorry for that, I'll test it properly and next time I hope there won't be problems at all.
I will try to cover any possible scenario.
Again sorry for the ping pong and for the bad tested code

**Pull request closed without merging**